sdk/rust/oak/src/lib.rs

```diff
@@ -299,6 +299,146 @@ pub fn node_create_with_label(
     result_from_status(status as i32, ())
 }
 
+/// Get the [`Label`] for the channel associated with the `handle`.
+pub fn channel_label(handle: Handle) -> Result<Label, OakStatus> {
+    let mut bytes = Vec::with_capacity(1024);
+    for resized in &[false, true] {
+        let mut actual_size: u32 = 0;
+        let status = OakStatus::from_i32(unsafe {
+            oak_abi::channel_label(
+                handle,
+                bytes.as_mut_ptr(),
+                bytes.capacity(),
+                &mut actual_size,
+            ) as i32
+        });
```

Comment:
Might be neater to have a `.ok_or(OakStatus::ErrInternal)?` here and drop the outer `match` below?

Reply:
(I imagine this is pre-existing and my fault in the first place, but I spot more simplifications now that I've made a [map](http://tinyurl.com/rust-transform))